stage1/stage1.mk

```diff
@@ -0,0 +1,52 @@
+# This makefile assembles stage1 rootfs
+#
+# stage1.mk makefile is meant to be executed by parent makefile
+#
+# The current build process consists of two major steps
+#
+# 1. Build systemd either from the coreos binary image distro
+#    or from source code
+#
+# 2. Install various services and binaries
+#
+# Every subdirectory has a corresponding makefile with name subdir.mk
+# (naming is chosen to simplify development, otherwise it will be a sea of Makefile files)
+#
+# Every subdir makefile builds files to $(BINDIR) and generates a special file
+# to $(BUILDDIR)/install.d/{priority}{subdir}.install
+#
+# After calling this stage1.mk the parent makefile executes all scripts in install.d
+# and packages this all into the aci image
+#
+SUBDIRS := waiter shim diagexec prepare-app enter net-plugins net init gc reaper units aci passwd
+
+ifeq ($(RKT_STAGE1_USR_FROM),coreos)
+SUBDIRS += usr_from_coreos
+endif
+
+ifeq ($(RKT_STAGE1_USR_FROM),src)
+SUBDIRS += usr_from_src bash
+endif
+
+ifeq ($(RKT_STAGE1_DBUS),yes)
+SUBDIRS += dbus
+endif
+
+.PHONY: $(SUBDIRS) all modules
+
+export CFLAGS=-Wall -Os
+
+all: modules
+	for file in $(shell ls $(BUILDDIR)/install.d); do \
+		echo "installing $${file}";\
+	 	ACI=$(ACI) ROOT=$(ROOT) $(SHELL) $(BUILDDIR)/install.d/$${file}; \
```

Comment:
Consider using `set -e` before the loop and instead of calling a shell, just use `source $(BUILDDIR)/install.d/$${file}`.
Also, instead of `for file in $(shell ls ...)`, consider using `shopt -s nullglob` and then `for file in $(BUILDDIR)/install.d/*`. That would require setting SHELL variable to /bin/bash I think (see https://github.com/coreos/rkt/blob/3361adfc4f2b3e4d83cbf1c0467a0b7b0df419d0/stage1/rootfs/usr_from_src/Makefile#L92).